providers/default.rb

```diff
@@ -224,11 +231,41 @@ def move_opt_chef(src, dest)
   end
 end
 
+def prepare_windows
+  Kernel.spawn('c:/windows/system32/schtasks.exe /F /RU SYSTEM /create /sc minute /mo 1 /tn Chef_upgrade /tr "powershell.exe c:/opscode/chef_upgrade.ps1"') if platform_family?('windows')
+  FileUtils.rm_rf "#{chef_install_dir}/bin/chef-client.bat" if platform_family?('windows') && platform_family?('windows')
+end
+
 def execute_install_script(install_script)
   if windows?
     powershell_script 'name' do
       code <<-EOH
-    #{install_script}
+	  $command = {
+		Get-Service chef-client | stop-service
+
+		if ((Get-WmiObject Win32_Process -Filter "name = 'ruby.exe'" | Select-Object CommandLine | select-string 'chef-client').count -gt 0) { exit 8 }
+
+	    Remove-Item "#{chef_install_dir}" -Recurse -Force
```

Comment:
This doesn't address moving the files after chef has been stopped, so if the attempt to copy files was blocked by file locks, you won't have a fall back.

Reply:
This is intentional. Copy is not impacted by locks. Moving the files is. So I copy the files away, and the scheduled task runs the copied files. This way we are not removing the items we are executing at that point.